### PR TITLE
Add multi screen mode

### PR DESCRIPTION
**Pull request by nihalsid**
- Adds multiscreen capabilities
- Monitors redis queue "screen_switch" every 1s for any message and switches
- modules in config is now a list of lists instead of a simple list, with each list being the modules on respective screens

### CODE REVIEW

File: js/electron.js [hunk 1]
```diff
@@ -101,4 +106,11 @@ app.on("activate", function() {
 // This starts all node helpers and starts the webserver.
 core.start(function(c) {
 	config = c;
+	totalScreens = config.modules.length;
```
Comment (jjedele) [inline js/electron.js:109]: Can't there be multiple modules on one screen?

Comment (jjedele) [reply js/electron.js:109]: Ahh, I think I understand. You made the module section an array.

File: js/screenswitch.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+var redis = require('redis');
+
+var ScreenSwitchHelper = function(){
```
Comment (jjedele) [inline js/screenswitch.js:3]: I think if we want to contribute this back to MagicMirror, Redis might be to specific a technology to invoke this mechanism. Can we maybe add another HTTP endpoint to the server that will trigger screen changes?

**Review by jjedele — APPROVED** [review]
Nice work. If we want to contribute it back to MagicMirror though we should fid a trigger mechanism that does not force users to install Redis I think.

**Comment by jjedele** [comment]
Also, it we want to contribute this back, we should adapt the sample config file and make it backwards-compatible to old configs. This should be pretty easy by just looking at the type of the `modules`part and put it in an array if it's a single object.

File: js/electron.js (hunk 1 above)
Comment (nihalsid) [reply js/electron.js:109]: Yes, the modules is an array now

**Comment by nihalsid** [comment]
Pretty good idea, making the config compatible with the old config. I'll make the changes. About making the thing technology independent, maybe this should be configurable in the "config.js", such that when you turn on multiscreens, you specify a multiscreen handler or something that manages how screens are switched. Lets talk about the ideas in person.

**Pull request merged**